analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -134,6 +134,12 @@ protected override ProgramState PreProcessSimple(SymbolicContext context)
         {
             return ProcessIndexerAccess(context.State, propertyReference);
         }
+        else if (operation.AsArgument() is { } argument
+            && context.State.ResolveCaptureAndUnwrapConversion(argument.Value).TrackedSymbol() is { } symbol
+            && context.State[symbol] is { } symbolValue)
+        {
+            return context.State.SetSymbolValue(symbol, symbolValue.WithoutConstraint<CollectionConstraint>());
```

Comment:
`WithoutConstraint` returns `SymbolicValue.Empty` if there would be just one. While `SetSymbolValue` expects `null` to remove the value from the state.

Let's update `WithoutConstraint` to return `null` in case there was just one constraint. It's used only in UTs so far, so there should be no harm doing so.